### PR TITLE
fix(npm): Migrate unpublishSafe presets

### LINKED ISSUE #7982: Nested presets not being migrated
**What Renovate type, platform and version are you using?**

Hosted app, but applies to any

**Describe the bug**

The `:unpublishSafe` preset was not being migrated from deeply nested configs.

**Relevant debug logs**

Example repo: https://github.com/nlibjs/nodetool/blob/master/package.json

It references a shared config, which in turn references `:unpublishSafe`. This was triggering a config warning issue:

```
DEBUG: Preset fetch error (repository=renovate-tests/nodetool)
       "preset": ":unpublishSafe",
       "err": {
         "message": "Cannot convert undefined or null to object",
         "stack": "TypeError: Cannot convert undefined or null to object\n    at Function.keys (<anonymous>)\n    at getPreset (/Users/rhys/src/renovate/lib/config/presets/index.ts:162:29)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at async resolveConfigPresets (/Users/rhys/src/renovate/lib/config/presets/index.ts:217:27)\n    at async Object.resolveConfigPresets (/Users/rhys/src/renovate/lib/config/presets/index.ts:247:30)\n    at async mergeRenovateConfig (/Users/rhys/src/renovate/lib/workers/repository/init/config.ts:190:3)\n    at async getRepoConfig (/Users/rhys/src/renovate/lib/workers/repository/init/config.ts:238:12)\n    at async initRepo (/Users/rhys/src/renovate/lib/workers/repository/init/index.ts:21:12)\n    at async Object.renovateRepository (/Users/rhys/src/renovate/lib/workers/repository/index.ts:39:14)\n    at async Object.start (/Users/rhys/src/renovate/lib/workers/global/index.ts:67:7)\n    at async /Users/rhys/src/renovate/lib/renovate.ts:10:22"
       }
 INFO: Throwing preset error (repository=renovate-tests/nodetool)
       "validationError": "undefined. Note: this is a *nested* preset so please contact the preset author if you are unable to fix it yourself."
 INFO: Repository has invalid config (repository=renovate-tests/nodetool)
       "error": {
         "validationError": "undefined. Note: this is a *nested* preset so please contact the preset author if you are unable to fix it yourself.",
         "message": "config-validation",
         "stack": "Error: config-validation\n    at resolveConfigPresets (/Users/rhys/src/renovate/lib/config/presets/index.ts:228:25)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"
       }
```

I needed to add back `unpublishSafe` to resolve it.

Although I resolved this manually, we should identify the root cause and fix.

**To Reproduce**

Fork the above repo

**Additional context**

Breaking changes made in v24

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

Migrate presets as well

## Context:

Closes #7982

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] Unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/migration.spec.ts [hunk 1]
```diff
@@ -497,7 +497,7 @@ describe('config/migration', () => {
       res = configMigration.migrateConfig(config);
       expect(res.isMigrated).toBe(true);
       expect(res.migratedConfig).toMatchObject({
-        extends: ['foo', ':unpublishSafe', 'bar', 'npm:unpublishSafe'],
+        extends: ['foo', 'npm:unpublishSafe', 'bar', 'npm:unpublishSafe'],
```
Comment (zharinov) [inline lib/config/migration.spec.ts:500]: I hope we're okay with duplication. Instead, we have to make implementation more complex.

File: lib/config/presets/internal/default.ts [hunk 2]
```diff
@@ -584,7 +584,4 @@ export const presets: Record<string, Preset> = {
     description: 'deprecated alias for config:js-lib',
     extends: ['config:js-lib'],
   },
-  unpublishSafe: {
-    extends: ['npm:unpublishSafe'],
-  },
```
Comment (zharinov) [inline lib/config/presets/internal/default.ts:589]: Optionally, we can just release migration, leaving this alias as long as needed

File: lib/config/migration.spec.ts (hunk 1 above)
Comment (rarkins) [reply lib/config/migration.spec.ts:500]: Ideally we should filter out duplicates, especially as we plan to turn this migration into a PR for users soon. Can we check that it's not already in the array before adding it?

Comment (zharinov) [reply lib/config/migration.spec.ts]: Okay, we're not aiming to fix user-defined duplicates, but now we won't make new ones

**Review by viceice — APPROVED** [review]

**Pull request merged**